Default expression for editing input form

wlorenzetti opened the pull request:
Closes: #390

Commit:
Create new utility function `explode_expression` to get metadata info from QGIS expression string.

Commit:
PEP8

Commit:
Fix missing import re.

Commit:
Fix return function.

Commit:
Typo

Commit:
Add `default_expression` property to `input`->``options` of API layer `config`.

wlorenzetti commented:
@volterra79 on this branch I add the new `default_expression` options property for input field property. Structure of `default_expresion` in relative issue #390.

volterra79 commented:
@wlorenzetti can you attach a project and relative data to test the code? Thanks

Commit:
Move `default_expression` property into `input`->`options`.

Commit:
Fix current_value regex.

Commit:
Add test for explode_expression function.

Commit:
Add test for API REST config layer. Fields with default value as expression.

volterra79 committed:
Client:
Pull https://github.com/g3w-suite/g3w-client/pull/234
commit: https://github.com/g3w-suite/g3w-client/pull/234/commits/49bee6dbc6d07d7ba5d900bfabfb2995c6bf1666

Commit:
Typo

Commit:
Fix regex for `current_value('field')` or `current_value("field")`.

Commit:
Remove default value by expression evaluation on editing commit.

wlorenzetti commented:
@leolami I just update dev.g3wsuite.it with this branch, please check and try it and review. Thanks

volterra79 committed:
Client:
Pull https://github.com/g3w-suite/g3w-client/pull/234
commit: https://github.com/g3w-suite/g3w-client/pull/234/commits/d831de49925cce08d71f4c9a1f16d5b704a64969

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
commit: https://github.com/g3w-suite/g3w-client/commit/32652a145326cf38eedf1c993ad979cce34877b3

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
commit: https://github.com/g3w-suite/g3w-client/commit/e4b7d78da0e3c1047db754ab570d54b9093af090

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
commit: https://github.com/g3w-suite/g3w-client/commit/e4b7d78da0e3c1047db754ab570d54b9093af090

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
commit: https://github.com/g3w-suite/g3w-client/commit/9bc065883dc40c0eb7c8bfbf45ebdc11c347dab4

Commit:
Fix current_value field name with spaces before and after field name declaration.

Commit:
Add property `apply_on_update` to `default_expression` options.

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
commit: https://github.com/g3w-suite/g3w-client/commit/5064c47870bbb0a6c02ac5b5bab97789336e0c76
editing: https://github.com/g3w-suite/g3w-client-plugin-editing/commit/612a496c34bada44edabfa1210884768cd932b13

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
commit: https://github.com/g3w-suite/g3w-client/commit/5064c47870bbb0a6c02ac5b5bab97789336e0c76
editing: https://github.com/g3w-suite/g3w-client-plugin-editing/commit/ee9ddb741568dd3ac7226302af052e28157f8071

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
commit: https://github.com/g3w-suite/g3w-client/commit/f7ef5bb89d92f58330d72dbf0427be6ae092f0d8
editing:
https://github.com/g3w-suite/g3w-client-plugin-editing/commit/7f2bf6fbedab6e812ba31c8f0271bc4df1dfbec8
https://github.com/g3w-suite/g3w-client-plugin-editing/commit/54417739f246b4a8c35e70474ce8eb9497487074

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
commit: https://github.com/g3w-suite/g3w-client/commit/96b71ab0b217141c03a7d2e4c5d6235316c19a9a

Commit:
Fix eval expression REST API result when result is a QVariant NULL.

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
commit: https://github.com/g3w-suite/g3w-client/commit/53300ab8e4adef8d7ea6cd796d034d6f4ec824fd
editing: https://github.com/g3w-suite/g3w-client-plugin-editing/commit/ce37014fef240fea4e1095571e2b836b879554e2

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
commit: https://github.com/g3w-suite/g3w-client/commit/cf1b15c040adda8fe8f1713dec1e59b12846f8ee
editing: https://github.com/g3w-suite/g3w-client-plugin-editing/commit/5f11153244b7007e6bcb420770aac592e49ddad6

Commit:
Add general scopes to expression_eval.

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

Commit:
Fix datetime fields on editing update. Porting from `dev` branch.

volterra79 committed:
Client:
branch https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

volterra79 committed:
Client:
editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

volterra79 committed:
Client:
branch: https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
Branch arpapiemonte is merge 3 pull request
https://github.com/g3w-suite/g3w-client/pull/234
https://github.com/g3w-suite/g3w-client/pull/207
https://github.com/g3w-suite/g3w-client/pull/146

editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

volterra79 committed:
Client:
branch: https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
Branch arpapiemonte is merge 3 pull request
https://github.com/g3w-suite/g3w-client/pull/234
https://github.com/g3w-suite/g3w-client/pull/207
https://github.com/g3w-suite/g3w-client/pull/146

editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

Commit:
Add parent management for eval expression.

volterra79 committed:
Client:
branch: https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
Branch arpapiemonte is merge 3 pull request
https://github.com/g3w-suite/g3w-client/pull/234
https://github.com/g3w-suite/g3w-client/pull/207
https://github.com/g3w-suite/g3w-client/pull/146

editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

Commit:
Refactoring remove field_espression_values.

volterra79 committed:
Client:
branch: https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
Branch arpapiemonte is merge 3 pull request
https://github.com/g3w-suite/g3w-client/pull/234
https://github.com/g3w-suite/g3w-client/pull/207
https://github.com/g3w-suite/g3w-client/pull/146

editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

Commit:
Fix date formatting for Vector API REST data and editing.

Commit:
Fix

volterra79 committed:
Client:
branch: https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
Branch arpapiemonte is merge 3 pull request
https://github.com/g3w-suite/g3w-client/pull/234
https://github.com/g3w-suite/g3w-client/pull/207
https://github.com/g3w-suite/g3w-client/pull/146

editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

volterra79 committed:
Merge branch 'default_expression' of https://github.com/g3w-suite/g3w-admin into default_expression

Commit:
Fix search for date datetime and time fields.

volterra79 committed:
Client:
branch: https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
Branch arpapiemonte is merge 3 pull request
https://github.com/g3w-suite/g3w-client/pull/234
https://github.com/g3w-suite/g3w-client/pull/207
https://github.com/g3w-suite/g3w-client/pull/146

editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

Commit:
Update comment.

Commit:
Fix date fields identify on editing and data vector API REST.

volterra79 committed:
Client:
branch: https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
Branch arpapiemonte is merge 3 pull request
https://github.com/g3w-suite/g3w-client/pull/234
https://github.com/g3w-suite/g3w-client/pull/207
https://github.com/g3w-suite/g3w-client/pull/146

editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

volterra79 committed:
Client:
branch: https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
Branch arpapiemonte is merge 3 pull request
https://github.com/g3w-suite/g3w-client/pull/234
https://github.com/g3w-suite/g3w-client/pull/207
https://github.com/g3w-suite/g3w-client/pull/146

editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

Commit:
Add `globalProjectLayer` scopes and `parentFormScope` to `QgsExpressionFilterBackend`

volterra79 committed:
Client:
branch: https://github.com/g3w-suite/g3w-client/tree/arpapiemonte
Branch arpapiemonte is merge 3 pull request
https://github.com/g3w-suite/g3w-client/pull/234
https://github.com/g3w-suite/g3w-client/pull/207
https://github.com/g3w-suite/g3w-client/pull/146

editing: https://github.com/g3w-suite/g3w-client-plugin-editing/tree/unique_values_handler

Commit:
Typo: fix expressionrule list url.

Commit:
Merge branch 'dev' into default_expression

# Conflicts:
#	g3w-admin/client/static/client/js/app.min.js
#	g3w-admin/editing/api/base/views.py
